### PR TITLE
[examples] Recommend using meldis where possible

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
For now, demos of hydroelastics or demos with camera images remain pinned to drake_visualizer.  Neither `geometry::Meshcat` nor `meldis` support that data (yet).

The problem in #16454 is what motivated some of the "realtime rate = 1.0" changes to README files here.  Some of our examples were running at crazy multiples of realtime, which overloaded the visualizer.  I've already fixed the overloading issue, but I still think it doesn't make sense to run short demos at dozens of multiples of realtime -- they are difficult to watch in that case.

Towards #16216.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452)

### CODE REVIEW

**Review by jwnimmer-tri — COMMENTED** [review]
I'll propose to share the feature review and platform, as follows:

+@sherm1 for **feature** review of these folders, please ...
+@sammy-tri for **platform** review of these folders, please ...

```
examples/acrobot/...
examples/multibody/...
examples/particles/...
examples/simple_gripper/...
multibody/plant/...
```

+@sammy-tri for **feature** review of these folders, please ...
+@sherm1 for **platform** review of these folders, please ...
```
examples/allegro_hand/...
examples/atlas/...
examples/compass_gait/...
examples/kinova_jaco_arm/...
examples/kuka_iiwa_arm/...
examples/planar_gripper/...
examples/rimless_wheel/...
```

My goal is to receive your review comments before the weekend, so that this can merge sometime on Monday (at the latest).

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-MunaCXA0hAjZnpGrrT2:bea063g)* status: LGTM missing from assignees sammy-tri(platform),sherm1(platform) (waiting on @sammy-tri and @sherm1)

**Review by sammy-tri — COMMENTED** [review]
Did my feature review section, will wait for sherm to feature review before doing platform.  It would be nice eventually to have the equivalent of `show_time.py` in meldis...  would be easier to figure out the rate a simulation is running at, anyway.

> Reviewed 7 of 23 files at r1.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-Muqq1BR8DKpCXdudFZw:b32e3w7)* status: LGTM missing from assignees sammy-tri(platform),sherm1(platform) (waiting on @sammy-tri and @sherm1)

**Review by sherm1 — COMMENTED** [review]
Feature :lgtm: for my assigned pieces, with a few btws.

> Reviewed 16 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-Mur8iym9NwnjswN2-z0:b5csbrm)* status: 2 unresolved discussions, LGTM missing from assignee sammy-tri(platform) (waiting on @jwnimmer-tri)

___
*[examples/simple_gripper/README.md, line 61 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16452#-Mur8Bc33xE6JeaFZ2jB:-Mur8Bc33xE6JeaFZ2jC:b-tmh8fw) ([raw file](https://github.com/robotlocomotion/drake/blob/0006863e48675d55dad39c71e9ba2135626d4d52/examples/simple_gripper/README.md#L61)):*
> ```Markdown
> ```
> ./bazel-bin/examples/simple_gripper/simple_gripper --simulation_time=10.0 \
>   --simulator_target_realtime_rate=1.0
> ```

BTW is adding the realtime_rate argument related to using meldis or just an opportunistic fix?
___
*[examples/simple_gripper/simple_gripper.cc, line 348 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16452#-Mur8V2GAujEazsPr3Ps:-Mur8V2GAujEazsPr3Pt:bkbcqbs) ([raw file](https://github.com/robotlocomotion/drake/blob/0006863e48675d55dad39c71e9ba2135626d4d52/examples/simple_gripper/simple_gripper.cc#L348)):*
> ```cpp
>       "scenario. SceneGraph is used for both visualization and contact "
>       "handling. "
>       "Launch meldis before running this example.");
> ```

BTW could combine the last two lines now.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452)* status: LGTM missing from assignee sammy-tri(platform) (waiting on @jwnimmer-tri)

___
*[examples/simple_gripper/README.md, line 61 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16452#-Mur8Bc33xE6JeaFZ2jB:-Mur93bQ1QtmpiknRiQe:b-phcgx0) ([raw file](https://github.com/robotlocomotion/drake/blob/0006863e48675d55dad39c71e9ba2135626d4d52/examples/simple_gripper/README.md#L61)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW is adding the realtime_rate argument related to using meldis or just an opportunistic fix?
</blockquote></details>

The PR overview text was my attempt to explain this.
___
*[examples/simple_gripper/simple_gripper.cc, line 348 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16452#-Mur8V2GAujEazsPr3Ps:-Mur9EIADCIeIDrqNGWk:bgqc9xx) ([raw file](https://github.com/robotlocomotion/drake/blob/0006863e48675d55dad39c71e9ba2135626d4d52/examples/simple_gripper/simple_gripper.cc#L348)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW could combine the last two lines now.
</blockquote></details>

OK There is an informal pattern in a lot of these files to have this (rote) sentence on its own line.  I'm satisfied to leave it alone.

**Comment by jwnimmer-tri** [comment]
> It would be nice eventually to have the equivalent of show_time ...

Agreed.  We've talked about that, but I don't think there is an issue yet, so I filed #16486 for tracking.

**Review by sherm1 — COMMENTED** [review]
Platform :lgtm: for the complementary pieces.

Two random thoughts, ill timed and misplaced: 
- "meldis" is not a very suggestive or memorable name. Would be nice if it were something like "visualizer" instead!
- it seems a shame to have to include `-- --open_window` repeatedly. Could that be the default?

> Reviewed 7 of 23 files at r1.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-MuranmVBK9B03h78Xql:bforzog)* status: LGTM missing from assignee sammy-tri(platform) (waiting on @jwnimmer-tri)

___
*[examples/simple_gripper/README.md, line 61 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16452#-Mur8Bc33xE6JeaFZ2jB:-MuraA9c6vzKaKsUsOvs:bv2c84k) ([raw file](https://github.com/robotlocomotion/drake/blob/0006863e48675d55dad39c71e9ba2135626d4d52/examples/simple_gripper/README.md#L61)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The PR overview text was my attempt to explain this.
</blockquote></details>

And of course you did a fine job of explaining it which I managed to forget about by the time I got here!

**Comment by jwnimmer-tri** [comment]
> "meldis" is not a very suggestive or memorable name. Would be nice if it were something like "visualizer" instead!

I am happy to use a better name, if anyone can suggest one.  Having a program named "drake-visualizer" and another one named "visualizer" seemed like a recipe for disaster, though.

> it seems a shame to have to include -- --open_window repeatedly. Could that be the default?

When I've written other similar kinds of tools in the past, people have complained about an auto-open feature, so I was a bit cautious here.  As I've been using it, there are definitely many times where I've kept the same browser tab open (so that it's placed correctly in my window manager), and then just re-loaded the page when I launch a new meldis server, instead of having meldis open a new window for me.  If we had a `--no-open` flag (defaulting to open, as you suggest) that workflow would still be possible, but I still tend to think that "open window" is the exception, rather than the rule.  We can revisit as more people start to use this; I'm open to debate.  (Another feature I'd like to see if for the web browser to reconnect automatically when I restart the server.  There's really no reason I should need to click the "reload" button.)

**Comment by sherm1** [comment]
> Having a program named "drake-visualizer" and another one named "visualizer" seemed like a recipe for disaster

Presumably the `drake-visualizer` program will be gone soon though, so any confusion would be short lived. And the instructions will only mention `visualizer` so new users wouldn't necessarily even know drake-visualizer was there.

An alternative would be `new-visualizer` but that has built in obsolescence.

Regarding `--open-window` I defer to your sense of the actual workflow which is just hypothetical for me.

**Comment by jwnimmer-tri** [comment]
> Presumably the drake-visualizer program will be gone soon though, ...

The `drake-visualizer` will be de-prioritized / have reduced prominence, but it will still exist on Focal until early 2024 (per #16215).

It's also the case that this LCM-driven display is probably not the final answer, either.  It's a nice compatibility shim to let us cut over quickly (especially on macOS), but it foregoes some of the nice two-way communication and interactivity that is possible if we were using meshcat natively.

In short, `meldis` is probably not the "one true visualizer" to the exclusion of all other mechanisms.  It's really just an LCM display server polyfill, to help smooth over the transition.

**Review by sherm1 — COMMENTED** [review]
+@xuchenhan-tri for platform shadow

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-MuryIlOEjmIdOfA7hoL:b-gvua7w)* status: LGTM missing from assignees sammy-tri(platform),xuchenhan-tri (waiting on @xuchenhan-tri)

**Review by xuchenhan-tri — COMMENTED** [review]
platform shadow :lgtm: 
I didn't go through the examples folder to check if the example should be retrofitted. Trusting the author and the two reviewers here.

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-MusJSyB-3qMxUy5jMTM:bllo6s7)* status: LGTM missing from assignee sammy-tri(platform) (waiting on @jwnimmer-tri)

**Comment by jwnimmer-tri** [comment]
@sammy-tri this ready for a final look (and lgtm emoji), when you have time.

**Review by sammy-tri — COMMENTED** [review]
:lgtm:

> Reviewed 16 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16452#-:-MuwXBKI4npZh--aqMxX:bnfp4nl)* status: :shipit: complete! all discussions resolved, LGTM from assignees sammy-tri(platform),sherm1(platform),xuchenhan-tri (waiting on @jwnimmer-tri)
